sim/priest/shadow/spriest.go

```diff
@@ -0,0 +1,246 @@
+						tickDamage *= 1 + float64(spriest.Talents.Darkness)*0.02
+						if spriest.Talents.Shadowform {
+							tickDamage *= 1.15
+						}
+					}
+					if tickDamage > 0 && spriest.VTSpell.DotInput.IsTicking(sim) {
+						s := stats.Stats{stats.Mana: tickDamage * 0.05}
+						if sim.Log != nil {
+							sim.Log("VT Regenerated %0f mana.\n", s[stats.Mana])
+						}
+						spriest.Party.AddStats(s)
+					}
+					return tickDamage
+				},
+				OnSpellHit: func(sim *core.Simulation, spellCast *core.SpellCast, spellEffect *core.SpellEffect) {
+					addShadowWeaving(sim, spellEffect)
+					if spellEffect.Damage > 0 && spriest.VTSpell.DotInput.IsTicking(sim) {
+						s := stats.Stats{stats.Mana: spellEffect.Damage * 0.05}
+						if sim.Log != nil {
+							sim.Log("VT Regenerated %0.1f mana.\n", s[stats.Mana])
+						}
+						spriest.Party.AddStats(s)
+					}
+
+					if spellCast.ActionID.SpellID == priest.SpellIDSWP || spellCast.ActionID.SpellID == priest.SpellIDVT || spellCast.ActionID.SpellID == priest.SpellIDMF {
+						spellEffect.Target.ReplaceAura(sim, core.Aura{
+							ID:      core.MiseryDebuffID,
+							Expires: sim.CurrentTime + misDur,
+							Name:    "Misery",
+							OnBeforeSpellHit: func(sim *core.Simulation, spellCast *core.SpellCast, spellEffect *core.SpellEffect) {
+								spellEffect.DamageMultiplier *= 1.05
+							},
+							OnPeriodicDamage: func(sim *core.Simulation, spellCast *core.SpellCast, spellEffect *core.SpellEffect, tickDamage float64) float64 {
+								return tickDamage * 1.05
+							},
+						})
+					}
+				},
+			}
+		})
+	}
+
+	return spriest
+}
+
+type ShadowPriest struct {
+	priest.Priest
+
+	swStacks float64
+
+	rotation proto.ShadowPriest_Rotation
+}
+
+func (spriest *ShadowPriest) GetPriest() *priest.Priest {
+	return &spriest.Priest
+}
+
+func (spriest *ShadowPriest) Reset(sim *core.Simulation) {
+	spriest.Priest.Reset(sim)
+	spriest.swStacks = 0
+}
+
+func (spriest *ShadowPriest) Act(sim *core.Simulation) time.Duration {
+	// Activate shared behaviors
+	target := sim.GetPrimaryTarget()
+	var spell *core.SimpleSpell
+	var wait time.Duration
+
+	timeForDots := sim.Duration-sim.CurrentTime > time.Second*12
+	if spriest.UseShadowfiend &&
+		spriest.CurrentMana()/spriest.MaxMana() < 0.5 &&
+		spriest.GetRemainingCD(priest.ShadowfiendCD, sim.CurrentTime) == 0 &&
+		!spriest.ShadowfiendSpell.DotInput.IsTicking(sim) {
+		spell = spriest.NewShadowfiend(sim, target)
+	} else if spriest.Talents.VampiricTouch && timeForDots && !spriest.VTSpell.DotInput.IsTicking(sim) {
+		spell = spriest.NewVT(sim, target)
+	} else if timeForDots && !spriest.SWPSpell.DotInput.IsTicking(sim) {
+		spell = spriest.NewSWP(sim, target)
+	} else if spriest.rotation.UseDevPlague && timeForDots && spriest.GetRemainingCD(priest.DPCooldownID, sim.CurrentTime) == 0 {
+		spell = spriest.NewDevouringPlague(sim, target)
+	} else if spriest.Talents.MindFlay {
+		mbcd := spriest.Character.GetRemainingCD(priest.MBCooldownID, sim.CurrentTime)
```

Comment:
Mind blast and SWD logic don't need to be in this MindFlay block, they can be done in their own blocks.

Reply:
was trying to avoid calculating remaining CDs on those unless we needed to... we need to reference those values below

Reply:
Ah ok. Then At least move the MindFlay talent check so it only applies to the mindflay stuff. Probably merge this whole block with the 'else' block after.